### ISSUE TITLE
"CA5390: Do Not Hard Code Encryption Key" does not work

### ISSUE THREAD

**Issue by avivanoff** [dotnet/roslyn-analyzers#2905]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.5 (Latest)

#### Diagnostic ID

CA5390

#### Repro steps

1. Create a simple console app.
2. Add FxCop analyzers and make sure CA5390 is enabled.
3. Add the following code:
```
using (var aes = Aes.Create())
{
    aes.Key = new Byte[] { 1, 2, 3 };
}

```

#### Expected behavior

CA5390 warning/error should be triggered.

#### Actual behavior

CA5390 warning/error is not triggered.

**Comment by dotpaul** [comment]
Hi @avivanoff, the example code should trigger CA5390, and it seems to trigger for me.
![image](https://user-images.githubusercontent.com/38924837/66161456-299bf600-e5e1-11e9-8ef0-9785bf6f7a03.png)

**Comment by avivanoff** [comment]
Strange. When I did a full rebuild, it started to work.

**Comment by mavasani** [comment]
@avivanoff I think that is likely because lot of security analyzers are build-only analyzers (compilation end based). They only run in IDE live analysis when you turn [enable full solution analysis](https://docs.microsoft.com/en-us/visualstudio/code-quality/how-to-enable-and-disable-full-solution-analysis-for-managed-code?view=vs-2019).

**Comment by avivanoff** [comment]
Actually, I believe the rule does have a problem. The following code is not marked:

```
internal static class Program
{
    private static readonly Byte[] _key = { 1, 2, 3 };

    private static void Main()
    {
        using (var aes = Aes.Create())
        {
            aes.Key = Program._key;
        }
    }
}

```

**Comment by dotpaul** [comment]
Thanks for reporting, @avivanoff !  We currently don't analyze field initializers, which is why that example doesn't produce a warning.  For this rule, I think we'll have to investigate how we can take field initializers into account, since I'm guessing initializing byte arrays is common enough to be worth analyzing.
